StreetHawk/Classes/Core/Publish/SHApp.m

```diff
@@ -1441,6 +1441,11 @@ - (void)sendModuleTags
         {
             [StreetHawk tagString:locationCurrent forKey:@"sh_module_location"];
             [[NSUserDefaults standardUserDefaults] setObject:locationCurrent forKey:@"sh_module_location"];
+            // set sh_location_denied as true when sh_module_location not be included
+            if ([locationCurrent compare:@"false"] == NSOrderedSame) {
+                [StreetHawk tagString:@"true" forKey:@"sh_location_denied"];
+                SHLog(@"sh_location_denied set as true due to sh location module not be included in this app");
+            }
```

Comment:
Same as push, this will be sent many times, even permission not change. 
In push you have a preference saving previous sent value, and only send tag when permission change. Should you do same?